generic-harvester/harvester/source/netcdf.py

```diff
@@ -0,0 +1,267 @@
+                    yield dict(zip(self.field_names, record_values))
+
+
+# Return a function that maps the variable values passed to the required output value
+# as specified in the associated mapping definition
+#
+# Performance sensitive!!  Applied to each row of measurements
+
+def _get_mapper(name, defn, variables):
+    if name not in variables:
+        return lambda values: None
+
+    variable = variables[name]
+    fill_value = variable[:].fill_value
+
+    return lambda values: values[name].item() if values[name].item() != fill_value else None
+
+
+# function used to determine whether a variable contains date/time values
+
+def _is_datetime(variable):
+    # date/time variables must include a 'units' attribute of the form '<time units> since <reference time>'
+    return hasattr(variable, 'units') and re.match(r'.* since .*', variable.units)
+
+
+# returns numpy arrays containing variable values - date/time variables are converted to UTC date/time arrays
+
+def _get_values(variable):
+    if _is_datetime(variable):
+        calendar = variable.calendar if hasattr(variable, 'calendar') else 'standard'
+        units = variable.units
+        dates = netCDF4.num2date(variable[:], units, calendar)
+        make_utc = np.vectorize(pendulum.timezone('UTC').convert)
+        return make_utc(dates)
+    else:
+        return variable[:]
+
+
+class NetcdfValueSource(object):
+    """
+    Class for iterating over variable values in a netcdf file
+
+    Each invocation of the iterator returns the id of the file, and the next set of values for variables in the file
+    requested in the supplied mapping as a dictionary
+    """
+
+    def __init__(self, netcdf_file, mapping):
+        self.netcdf_file = netcdf_file
+        self.mapping = mapping
+        self.field_names = ("file_id",) + tuple(self.mapping["fields"].keys())
+
+    def records(self):
+        """
+        :return: iterator returning each record as a dictionary of field/value pairs
+        """
+
+        with netCDF4.Dataset(self.netcdf_file.src_path) as dataset:
+            dataset.set_auto_scale(True)
+            variables = dict((name, variable) for name, variable in dataset.variables.items()
+                             if variable.dimensions == tuple(self.mapping["dimensions"]))
+            field_mappers = [_get_mapper(name, defn, variables) for name, defn in self.mapping["fields"].items()]
+            file_id = self.netcdf_file.id
+            # iterate through each variables values - can be millions so performance sensitive!
+            iterators = {name: np.nditer(_get_values(variable), flags=["refs_ok"]) for name, variable in variables.items()}
+            first_iterator = list(iterators.values())[0]
+            for index in range(*first_iterator.iterrange):
+                variable_values = {name: next(iterator) for name, iterator in iterators.items()}
+                record_values = [file_id] + [map(variable_values) for map in field_mappers]
```

Comment:
`map` is a reserved word, so worth renaming. Also, it's generally acceptable (or even idiomatic) to use single character names in comprehensions )(e.g. `m` here), but not outside comprehensions, where readability is more important.